src/build-tools/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Provides access to tools required in build scripts.
+//!
+//! For example, many crates have a build script that depends on the Protobuf
+//! compiler, `protoc`. If we're building with Cargo we'll bootstrap `protoc`
+//! by compiling it with [`protobuf-src`], but if we're building with Bazel
+//! then we'll use the version of `protoc` included in the runfiles.
+
+use cfg_if::cfg_if;
+use std::path::PathBuf;
+
+// Note: This crate's BUILD.bazel compiles with the rustc flag `--cfg=bazel`.
+
+// Runfiles are a Bazel concept, they're a way to provide files at execution
+// time. This dependency is provided only by the Bazel build.
+#[cfg(bazel)]
+extern crate runfiles;
+
+/// Returns the path to `protoc`.
+///
+/// Looks for `protoc` in the following places:
+///
+/// * Bazel runfiles, if we're building with Bazel.
+/// * Bootstraps `protoc` via protobuf-src, if default features are enabled.
+/// * `PROTOC` environment variable, if it's set.
+/// * The system's `$PATH`, via [`which`].
+///
+/// If `protoc` can't be found then this function will panic.
+pub fn protoc() -> PathBuf {
+    cfg_if! {
+        if #[cfg(bazel)] {
+            let r = runfiles::Runfiles::create().unwrap();
```

Comment:
is this a bazel-provided thing?

Reply:
Yeah this is provided by the [Bazel rules_rust](https://github.com/bazelbuild/rules_rust/tree/main/tools/runfiles) project